packages/react-internal/src/components/SearchBox/SearchBox.base.tsx

```diff
@@ -130,28 +132,32 @@ export const SearchBoxBase: React.FunctionComponent<ISearchBoxProps> = React.for
   const onKeyDown = (ev: React.KeyboardEvent<HTMLInputElement>) => {
     switch (ev.which) {
       case KeyCodes.escape:
-        onEscape?.(ev);
-        if (!ev.defaultPrevented) {
+        customOnEscape?.(ev);
+        // Only call onClear if the search box has a value to clear. Otherwise, allow the Esc key
+        // to propagate from the empty search box to a parent element such as a dialog, etc.
+        if (value && !ev.defaultPrevented) {
           onClear(ev);
         }
         break;
+
       case KeyCodes.enter:
-        if (onSearch) {
-          onSearch(value);
-          break;
+        if (customOnSearch) {
+          customOnSearch(value);
+          ev.preventDefault();
+          ev.stopPropagation();
         }
-        // if we don't handle the enter press then we shouldn't prevent default
-        return;
+        break;
+
       default:
+        // REVIEW: Why aren't we calling customOnKeyDown for Escape or Enter?
```

Comment:
My guess would be it's because we already have specialized customizable callbacks for those keys specifically. But maybe we should at least update the `onKeyDown` documentation to note that it's not called for those keys.